tests/rados/test_mon_pg_warn.py

```diff
@@ -0,0 +1,145 @@
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_obj)
+    pool_obj = PoolFunctions(node=cephadm)
+    pool_name = "test_mon_pg_warn"
+    log.info(
+        "Running test case to verify auto increment of 'mon_pg_warn_max_object_skew' parameter with PG auto-scaling"
+    )
+
+    def mon_pg_warn_health_warn(should_contain: bool):
+        # check health warning for "MANY_OBJECTS_PER_PG"
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=125)
+        while datetime.datetime.now() < end_time:
+            try:
+                health_detail, _ = cephadm.shell(args=["ceph health detail"])
+                log.info(f"Health warning: \n {health_detail}")
+                if should_contain:
+                    assert "[WRN] MANY_OBJECTS_PER_PG:" in health_detail
+                else:
+                    assert "[WRN] MANY_OBJECTS_PER_PG:" not in health_detail
+                break
+            except AssertionError:
+                time.sleep(30)
+                if datetime.datetime.now() >= end_time:
+                    raise
+
+    for test_type in ["manual_increment", "pg_autoscale"]:
+        try:
+            # fetch the default value of MON parameter 'mon_pg_warn_max_object_skew'
+            default_mon_pg_warn = float(
+                mon_obj.get_config(section="mon", param="mon_pg_warn_max_object_skew")
+            )
+
+            if not default_mon_pg_warn == 10.000000:
+                err = (
+                    "Default value of mon parameter 'mon_pg_warn_max_object_skew' has changed. "
+                    f"Expected: 10.000000 | Actual: {default_mon_pg_warn}"
+                )
+                log.error(err)
+                raise AssertionError(err)
+
+            rados_obj.configure_pg_autoscaler(**{"default_mode": "warn"})
+            # create a replicated pool with single pg
+            assert rados_obj.create_pool(
+                pool_name=pool_name, **{"pg_num": 1, "pgp_num": 1, "pg_num_max": 1}
+            )
+
+            # write 10000 objects to the pool to trigger the warning
+            bench_config = {
+                "rados_write_duration": 300,
+                "byte_size": "64KB",
+                "max_objs": 10000,
+            }
+            assert rados_obj.bench_write(pool_name=pool_name, **bench_config)
+
+            # check health warning for "MANY_OBJECTS_PER_PG"
+            mon_pg_warn_health_warn(should_contain=True)
+
+            if test_type == "pg_autoscale":
+                # enable PG autoscaling
+                mon_obj.remove_config(
+                    section="global", name="osd_pool_default_pg_autoscale_mode"
+                )
+                rados_obj.set_pool_property(
+                    pool=pool_name, props="pg_autoscale_mode", value="on"
+                )
+                rados_obj.set_pool_property(
```

Comment:
post setting the pg_autoscale mode on, Were the PGs of the pool not autoscaled automatically? 

I believe this should happen.

Not sure if this has to do anything with the param : `pg_num_max`.